### PR TITLE
Allow build on FreeBSD

### PR DESCRIPTION
**Pull request by herrhotzenplotz**
These are some fixes for making contour compile on FreeBSD. Mostly
things where Linux was assumed.

Build with autogen.sh does not work because of the broken shebang (which I didn't bother fixing).
Thus build manually with cmake:

$ cmake -G'Unix Makefiles' -Bbuild .

Everything seems fine so far, except changing the font. That causes
a floating point exception. I may do some further debugging to try
and figure out what is going on there.

I don't know if the change in UnixPty.h:108 causes any trouble because of const correctness.
The prototype of openpty on FreeBSD is : 
```
 int
 openpty(int *amaster, int *aslave, char *name, struct termios *termp,
         struct winsize *winp);
```
(see https://www.freebsd.org/cgi/man.cgi?query=openpty&apropos=0&sektion=0&manpath=FreeBSD+13.0-RELEASE+and+Ports&arch=default&format=html ). The compiler complains about the discarded const which causes the candidade to be disregarded.

In case the changes get merged and the floating point exception fixed, I may bring contour to the FreeBSD Ports.

### CODE REVIEW

**Comment by NTBBloodbath** [comment]
Hey, just as a note if you want to fix the autogen script shebang to be more portable between *nix systems (including BSD).

There are two options IIRC (the first one being better)
1. You could use `#!/bin/sh` as the shebang (this one will find a mostly POSIX-compliant shell on any standard Unix system)
2. You could use `#!/usr/bin/env bash` as the shebang

**Review by uspasojevic96 — CHANGES_REQUESTED** [review]

**Comment by christianparpart** [comment]
Oh @herrhotzenplotz that is sooooo nice of you in getting Contour runnable on FreeBSD. That is actually really awesome work. Please give me some time to respectfully review your work. I'm looking forward to it.

**Comment by christianparpart** [comment]
> You could use #!/bin/sh as the shebang (this one will find a mostly POSIX-compliant shell on any standard Unix system)

I think at least as far as compatibility is concerned, `/bin/sh` would be the way to go.

NB: I'd love to get FreeBSD build tests into the Github CI. But that doesn't have to be done as part of this PR, this may very well start with a ticket on its own. :)

File: src/terminal/pty/UnixPty.h [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 #if defined(__APPLE__)
 #include <util.h>
+#elif defined(__FreeBSD__)
```
Comment (christianparpart) [inline src/terminal/pty/UnixPty.h:24]: Empty elif-branch can be eliminated.

File: src/terminal/Process.cpp [hunk 2]
```diff
@@ -29,7 +29,9 @@
 #include <unordered_map>
 
 #if !defined(_WIN32)
+#if !defined(__FreeBSD__)
```
Comment (christianparpart) [inline src/terminal/Process.cpp:32]: double `#if !...` may be a little weird to look at / review. But no need to change, I can do that later.

**Review by christianparpart — COMMENTED** [review]
The empty `#if...`-branches do not need to exist. Apart from this: LGTM. Many thanks @herrhotzenplotz :+1:

File: src/terminal/pty/UnixPty.h (hunk 1 above)
Comment (herrhotzenplotz) [reply src/terminal/pty/UnixPty.h:24]: Issue is that pty.h does not exist on FreeBSD. If it is only a thing on Linux, then we may wanna change this into a ifdef for Linux.

File: src/terminal/Process.h [hunk 3]
```diff
@@ -33,6 +33,7 @@
 #include <libproc.h>
 #include <terminal/pty/UnixPty.h>
 #include <util.h>
+#elif defined(__FreeBSD__)
```
[the remark this answers is not in the document]
Comment (herrhotzenplotz) [reply src/terminal/Process.h:36]: Same issue as below

**Comment by herrhotzenplotz** [comment]
> Hey, just as a note if you want to fix the autogen script shebang to be more portable between *nix systems (including BSD).
> 
> There are two options IIRC (the first one being better)
> 
>     1. You could use `#!/bin/sh` as the shebang (this one will find a mostly POSIX-compliant shell on any standard Unix system)
> 
>     2. You could use `#!/usr/bin/env bash` as the shebang

Indeed that is the usual way to go about this problem. However, the script is already filled with bashims which I didn't want to fix. I guess for anything not GNU based we should simply ignore this script. Otherwise this must be rewritten in POSIX sh or something else that fits. Depending on bash is not really that appreciated on *BSD as you may guess :smile:

**Review by christianparpart — APPROVED** [review]
@herrhotzenplotz I'm gladly merging this. I think the autogen.sh script is not important as it's meant to be used by devs (not by automated scripts). I can fix that very well later (or someone else, that would like to volunteer, in its own PR).

Do you agree with a merge in this state or do you have anything else planned to be pushed?

**Comment by christianparpart** [comment]
Oh ehm, @herrhotzenplotz: Would you mind sharing me your installation prerequisites for FreeBSD? I'd like to try adding this to the CI for automated build testing.

**Comment by herrhotzenplotz** [comment]
> Do you agree with a merge in this state or do you have anything else planned to be pushed?

I'm looking into the floating point exceptions right now. But that is another PR. It can be merged like this now.

**Comment by herrhotzenplotz** [comment]
> Oh ehm, @herrhotzenplotz: Would you mind sharing me your installation prerequisites for FreeBSD? I'd like to try adding this to the CI for automated build testing.

I will write a ports Makefile and test it in poudriere. that way I can ensure the minimal set of dependencies that make it build and work properly. I'll let you know when it works! :smiley:

**Comment by herrhotzenplotz** [comment]
After some hack and patch and so on, I managed to produce a ports Makefile. To be honest, the CMake based build system is kind of a show stopper. Especially the CPM package manager (which I have never heard of before) is something that will cause problems on several packaging systems because it tries to download dependencies in the config/build stage. This is not allowed and causes failures in e.g. poudriere which is used to build the repositories for FreeBSD.
There are also lots of link problems because the correct library paths for the link stage are missing entirely. Probably a fix of the `target_link_libraries()` is needed.

The current ports shar including Makefile and patches etc. (not really polished and cleaned up, but it works) is at: https://gist.githubusercontent.com/herrhotzenplotz/4d545fce05643ee845dbfb18c0ec20d9/raw/24e14447bf190c7d767a2369b5bdd3e13fc21778/x11-contour.shar

The current build dependencies of it are: `pkg catch microsoft-gsl range-v3 cmake ninja ncurses pkgconf qt5-buildtools qt5-qmake libfmt fontconfig freetype2 harfbuzz yaml-cpp qt5-core qt5-gui qt5-network qt5-widgets`

You can use `pkg install -y <insert pkgs here>` to install them on the target machine.

I would suggest to use the same build flags for CMake as in the ports Makefile as this will make having an official port in FreeBSD easier.

**Comment by uspasojevic96** [comment]
@christianparpart aaaand CPM bites us in the ass :)

**Comment by christianparpart** [comment]
> @christianparpart aaaand CPM bites us in the ass :)

haha!

<details>
<summary>the link inlined just in case it goes wonka</summary>

```sh
# This is a shell archive.  Save it in a file, remove anything before
# this line, and then unpack it by entering "sh file".  Note, it may
# create directories; files and directories will be owned by you and
# have default permissions.
#
# This archive contains:
#
#	./Makefile
#	./distinfo
#	./files/patch-src_terminal__renderer_CMakeLists.txt
#	./files/patch-cmake_ThirdParties.cmake
#	./files/patch-src_crispy_CMakeLists.txt
#	./files/patch-src_terminal_CMakeLists.txt
#	./files/patch-CMakeLists.txt
#	./files/patch-src_contour_CMakeLists.txt
#	./files/patch-src_text__shaper_CMakeLists.txt
#	./pkg-descr
#
echo x - ./Makefile
sed 's/^X//' >./Makefile << '464ef7c6571951809b23e262cbe26e19'
XPORTNAME=	contour
XDISTVERSION=	v0.2.2.175
XCATEGORIES=	x11
X
XMAINTAINER=	nsonack@outlook.com
XCOMMENT=	Modern Terminal Emulator
XLICENSE=	APACHE20
XLICENSE_FILE=	${WRKSRC}/LICENSE.txt
X
XUSE_GITHUB=	yes
XGH_ACCOUNT=	contour-terminal
XGH_TAGNAME=	c2d03221
XGH_TUPLE=	contour-terminal:libunicode:3c59a3a:contour/libunicode
X
XBUILD_DEPENDS=	catch>0:devel/catch \
X		microsoft-gsl>0:devel/microsoft-gsl \
X		range-v3>0:devel/range-v3
XLIB_DEPENDS=	libfmt.so:devel/libfmt \
X		libfontconfig.so:x11-fonts/fontconfig \
X		libfreetype.so:print/freetype2 \
X		libharfbuzz.so:print/harfbuzz \
X		libyaml-cpp.so:devel/yaml-cpp
XUSES=		cmake ncurses:port pkgconfig qt:5
X
XUSE_QT=		core gui network widgets buildtools_build qmake_build
X
XCMAKE_ARGS+=	-DCONTOUR_TESTING=OFF \
X		-DCPM_LOCAL_PACKAGES_ONLY:BOOL=true \
X		-DCPM_USE_LOCAL_PACKAGES:BOOL=true \
X		-DCRISPY_TESTING=OFF \
X		-DLIBTERMINAL_TESTING=OFF
XPLIST_FILES=	bin/contour \
X		share/applications/contour.desktop \
X		share/kservices5/ServiceMenus/contour-run.desktop \
X		share/pixmaps/contour.png \
X		share/terminfo/c/contour \
X		share/terminfo/c/contour-latest
X
X.include <bsd.port.mk>
464ef7c6571951809b23e262cbe26e19
echo x - ./distinfo
sed 's/^X//' >./distinfo << 'aec8793adfd5b7c3f00e62ae41a09ac2'
XTIMESTAMP = 1639121916
XSHA256 (contour-terminal-contour-v0.2.2.175-c2d03221_GH0.tar.gz) = e9b5663dd3dc072694fed818101aa3322feb85772ce1965a959e09f2d4d21ec7
XSIZE (contour-terminal-contour-v0.2.2.175-c2d03221_GH0.tar.gz) = 2962008
XSHA256 (contour-terminal-libunicode-3c59a3a_GH0.tar.gz) = c5614014c155c9ba09e79a2453cd34c068ee9209bd6d928dbac8ee62b52e263c
XSIZE (contour-terminal-libunicode-3c59a3a_GH0.tar.gz) = 600886
aec8793adfd5b7c3f00e62ae41a09ac2
echo x - ./files/patch-src_terminal__renderer_CMakeLists.txt
sed 's/^X//' >./files/patch-src_terminal__renderer_CMakeLists.txt << 'e056099ea07c0b2825808d1cfa4f794d'
X--- src/terminal_renderer/CMakeLists.txt.orig	2021-12-09 19:17:13 UTC
X+++ src/terminal_renderer/CMakeLists.txt
X@@ -13,4 +13,4 @@ add_library(terminal_renderer STATIC
X )
X 
X target_include_directories(terminal_renderer PUBLIC ${PROJECT_SOURCE_DIR}/src ${CMAKE_SOURCE_DIR}/src)
X-target_link_libraries(terminal_renderer PUBLIC terminal crispy::core text_shaper range-v3)
X+target_link_libraries(terminal_renderer PUBLIC terminal crispy::core text_shaper ${RANGE_V3_LIBRARIES})
e056099ea07c0b2825808d1cfa4f794d
echo x - ./files/patch-cmake_ThirdParties.cmake
sed 's/^X//' >./files/patch-cmake_ThirdParties.cmake << '25f337d87532c3a752bfa2ae612fba64'
X--- cmake/ThirdParties.cmake.orig	2021-12-09 19:17:13 UTC
X+++ cmake/ThirdParties.cmake
X@@ -4,12 +4,12 @@ set(CPM_DOWNLOAD_ALL ON)
X set(3rdparty_catch2_version       "b9baae6d938133ff2fdc938053e88ccf51dd3afe" CACHE STRING "catch2: commit hash")
X set(3rdparty_fmt_version          "a3ab36c80399f13e05ff2ff142df58fa9f3fd103" CACHE STRING "fmt: commit hash")
X set(3rdparty_freetype_version     "cff026d41599945498044d2f4dcc0e610ffb6929" CACHE STRING "freetype: commit hash")
X-set(3rdparty_gsl_version          "e0880931ae5885eb988d1a8a57acf8bc2b8dacda" CACHE STRING "GSL: commit hash")
X+set(3rdparty_gsl_version          "" CACHE STRING "GSL: commit hash")
X set(3rdparty_harfbuzz_version     "720ab0883b4bd7daa32a3c46031a9d8adb5c8a5f" CACHE STRING "harfbuzz: commit hash")
X set(3rdparty_libunicode_version   "3c59a3a0eb0c57c3081d48ea77c22809ac7c2d6e" CACHE STRING "libunicode: commit hash")
X set(3rdparty_mimalloc_version     "0be71a2cac17062bd8913cbd272c472a44331b7f" CACHE STRING "mimalloc: commit hash")
X-set(3rdparty_range_v3_version     "83783f578e0e6666d68a3bf17b0038a80e62530e" CACHE STRING "range_v3: commit hash")
X-set(3rdparty_yaml_cpp_version     "1713859b054b0a7fd867a59905dfbb0d3f774d54" CACHE STRING "yaml-cpp: commit hash")
X+set(3rdparty_range_v3_version     "" CACHE STRING "range_v3: commit hash")
X+set(3rdparty_yaml_cpp_version     "" CACHE STRING "yaml-cpp: commit hash")
X set(3rdparty_termbenchpro_version "513cc8a7eb15b3b1a7940816944056d168d2c22a" CACHE STRING "termbench-pro: commit hash")
X 
X if(CONTOUR_TESTING OR CRISPY_TESTING OR LIBTERMINAL_TESTING)
X@@ -36,7 +36,7 @@ CPMAddPackage(
X )
X 
X CPMAddPackage(
X-  NAME range_v3
X+  NAME range-v3
X   GITHUB_REPOSITORY "ericniebler/range-v3"
X   GIT_TAG ${3rdparty_range_v3_version}
X   EXCLUDE_FROM_ALL YES
X@@ -49,33 +49,31 @@ set(YAML_CPP_INSTALL OFF CACHE INTERNAL "")
X set(YAML_BUILD_SHARED_LIBS OFF CACHE INTERNAL "")
X 
X CPMAddPackage(
X-  NAME yaml_cpp
X+  NAME yaml-cpp
X   GITHUB_REPOSITORY "jbeder/yaml-cpp"
X   GIT_TAG ${3rdparty_yaml_cpp_version}
X   EXCLUDE_FROM_ALL YES
X )
X 
X CPMAddPackage(
X-  NAME GSL
X-  GITHUB_REPOSITORY "microsoft/GSL"
X-  GIT_TAG ${3rdparty_gsl_version}
X+  NAME Microsoft.GSL
X   EXCLUDE_FROM_ALL YES
X )
X 
X-CPMAddPackage(
X-  NAME termbenchpro
X-  GITHUB_REPOSITORY "contour-terminal/termbench-pro"
X-  GIT_TAG ${3rdparty_termbenchpro_version}
X-  EXCLUDE_FROM_ALL YES
X-)
X+# CPMAddPackage(
X+#   NAME termbenchpro
X+#   GITHUB_REPOSITORY "contour-terminal/termbench-pro"
X+#   GIT_TAG ${3rdparty_termbenchpro_version}
X+#   EXCLUDE_FROM_ALL YES
X+# )
X 
X-CPMAddPackage(
X-  NAME libunicode
X-  GITHUB_REPOSITORY "contour-terminal/libunicode"
X-  GIT_TAG ${3rdparty_libunicode_version}
X-  EXCLUDE_FROM_ALL YES
X-)
X-
X+# CPMAddPackage(
X+#   NAME libunicode
X+#   GITHUB_REPOSITORY "contour-terminal/libunicode"
X+#   GIT_TAG ${3rdparty_libunicode_version}
X+#   EXCLUDE_FROM_ALL YES
X+# )
X+ 
X if(CONTOUR_BUILD_WITH_MIMALLOC)
X   set(MI_BUILD_SHARED OFF CACHE INTERNAL "")
X   set(MI_BUILD_TESTS OFF CACHE INTERNAL "")
25f337d87532c3a752bfa2ae612fba64
echo x - ./files/patch-src_crispy_CMakeLists.txt
sed 's/^X//' >./files/patch-src_crispy_CMakeLists.txt << '10c5c657798b9aa8f0ddcaecae091688'
X--- src/crispy/CMakeLists.txt.orig	2021-12-09 19:17:13 UTC
X+++ src/crispy/CMakeLists.txt
X@@ -27,7 +27,7 @@ set(crispy_SOURCES
X add_library(crispy-core ${crispy_SOURCES})
X add_library(crispy::core ALIAS crispy-core)
X 
X-set(CRISPY_CORE_LIBS range-v3 fmt::fmt-header-only unicode::core GSL)
X+set(CRISPY_CORE_LIBS fmt unicode::core)
X if(${USING_BOOST_FILESYSTEM})
X     target_compile_definitions(crispy-core PUBLIC USING_BOOST_FILESYSTEM=1)
X     list(APPEND CRISPY_CORE_LIBS Boost::filesystem)
10c5c657798b9aa8f0ddcaecae091688
echo x - ./files/patch-src_terminal_CMakeLists.txt
sed 's/^X//' >./files/patch-src_terminal_CMakeLists.txt << '0bfe524cfcb9d2c838c9f77dd19b5f0d'
X--- src/terminal/CMakeLists.txt.orig	2021-12-09 19:17:13 UTC
X+++ src/terminal/CMakeLists.txt
X@@ -79,7 +79,7 @@ set(terminal_SOURCES
X     primitives.cpp
X )
X 
X-set(LIBTERMINAL_LIBRARIES crispy::core fmt::fmt-header-only range-v3 Threads::Threads GSL)
X+set(LIBTERMINAL_LIBRARIES crispy::core Threads::Threads)
X if(UNIX)
X     list(APPEND LIBTERMINAL_LIBRARIES util)
X     list(APPEND terminal_SOURCES pty/UnixPty.cpp)
0bfe524cfcb9d2c838c9f77dd19b5f0d
echo x - ./files/patch-CMakeLists.txt
sed 's/^X//' >./files/patch-CMakeLists.txt << 'd16764cc78e8861b85cf994e3d0e2c58'
X--- CMakeLists.txt.orig	2021-12-10 08:40:02 UTC
X+++ CMakeLists.txt
X@@ -77,6 +77,7 @@ endif()
X 
X # ----------------------------------------------------------------------------
X add_subdirectory(src)
X+add_subdirectory(libunicode)
X 
X include(ThirdParties)
X 
d16764cc78e8861b85cf994e3d0e2c58
echo x - ./files/patch-src_contour_CMakeLists.txt
sed 's/^X//' >./files/patch-src_contour_CMakeLists.txt << '206d84c61f8029eda279eddab30b5afa'
X--- src/contour/CMakeLists.txt.orig	2021-12-09 19:17:13 UTC
X+++ src/contour/CMakeLists.txt
X@@ -110,6 +110,7 @@ elseif(APPLE)
X     )
X endif()
X 
X+target_link_options(contour PRIVATE -L$ENV{LOCALBASE}/lib)
X target_link_libraries(contour terminal yaml-cpp)
X if(Boost_FILESYSTEM_FOUND)
X     target_include_directories(contour PRIVATE ${Boost_INCLUDE_DIRS})
X@@ -292,10 +293,6 @@ else()
X     install(FILES "contour.desktop" DESTINATION "${CMAKE_INSTALL_DATADIR}/applications")
X     install(FILES "contour-run.desktop" DESTINATION "${CMAKE_INSTALL_DATADIR}/kservices5/ServiceMenus")
X     install(FILES "res/images/contour-logo-256.png" DESTINATION "${CMAKE_INSTALL_DATADIR}/pixmaps" RENAME "contour.png")
X-    install(FILES "shell-integration.zsh" DESTINATION "${CMAKE_INSTALL_DATADIR}/contour")
X-    install(FILES "${PROJECT_SOURCE_DIR}/Changelog.md" DESTINATION "${CMAKE_INSTALL_DATADIR}/contour")
X-    install(FILES "${PROJECT_SOURCE_DIR}/LICENSE.txt" DESTINATION "${CMAKE_INSTALL_DATADIR}/contour")
X-    install(FILES "${PROJECT_SOURCE_DIR}/README.md" DESTINATION "${CMAKE_INSTALL_DATADIR}/contour")
X     install(DIRECTORY "${terminfo_basedir}" DESTINATION "${CMAKE_INSTALL_DATADIR}")
X     # TODO: install Changelog.md to /usr/share/doc/contour/Changelog.md(.gz)?
X endif()
206d84c61f8029eda279eddab30b5afa
echo x - ./files/patch-src_text__shaper_CMakeLists.txt
sed 's/^X//' >./files/patch-src_text__shaper_CMakeLists.txt << '8d0cb831df43ab64d2bc8805b0d33d6a'
X--- src/text_shaper/CMakeLists.txt.orig	2021-12-09 19:17:13 UTC
X+++ src/text_shaper/CMakeLists.txt
X@@ -19,9 +19,7 @@ endif()
X add_library(text_shaper STATIC ${text_shaper_SRC})
X 
X set(TEXT_SHAPER_LIBS unicode::core)
X-list(APPEND TEXT_SHAPER_LIBS fmt::fmt-header-only)
X-list(APPEND TEXT_SHAPER_LIBS range-v3)
X-list(APPEND TEXT_SHAPER_LIBS GSL)
X+list(APPEND TEXT_SHAPER_LIBS fmt)
X 
X if(APPLE)
X     enable_language(OBJCXX)
8d0cb831df43ab64d2bc8805b0d33d6a
echo x - ./pkg-descr
sed 's/^X//' >./pkg-descr << '0e313719ff012ceaf97a028d51caa8f5'
Xcontour is a modern terminal emulator, for everyday use. It is
Xaiming for power users with a modern feature mindset.
X
XWWW: https://github.com/contour-terminal/contour/
0e313719ff012ceaf97a028d51caa8f5
exit
```
</details>

@herrhotzenplotz ok, I am really so sorry for the pain of CMake/CPM. I chose CMake because it's the most widely supported build system that also includes Windows (Visual Studio), as I'd like to retain easy buildability on Windows platform, too.

So you basically got rid of CPM to make it work? I think I must get a FreeBSD VM locally and see if I can repro that and maybe we can get a fix. 

**Just to be dog-clear:** So removing CPM was fixing the `SIGFPE`?

**Comment by herrhotzenplotz** [comment]
> **Just to be dog-clear:** So removing CPM was fixing the `SIGFPE`?

It didn't fix the FPE.

**Comment by christianparpart** [comment]
> Especially the CPM package manager (which I have never heard of before) is something that will cause problems on several packaging systems because it tries to download dependencies in the config/build stage. 

To be honest, i'd prefer all deps to be sysdeps (as in: installed by the system package manager before). Except `libunicode`, which is a helper-project as part of Contour, which isn't as prominent as yaml-cpp or  range-v3. So libunicode could use git submodules.

The whole purpose of the configure-stage downloading of dependencies is to make it easier for the users to compile from source (and for the devs, but yeah :D).

But now, thinking about it. We already have `./scripts/install-deps.sh` that is - based on your platform - installing system dependencies. We could actually Add FreeBSD support in this script (1.), and 2., we could then `git clone` into a `./_deps/` or alike all remaining deps that do NOT have a system package yet (like libunicode, most likely).

I hope I'm not bugging you too much @herrhotzenplotz with this, so just tell me to STFU otherwise :)

Would this script-based approach work for you (as in: FreeBSD)? I can just Google around to see if range-v3/yaml-cpp/Catch2 deps are available as Port or not.

**Comment by herrhotzenplotz** [comment]
> Would this script-based approach work for you (as in: FreeBSD)? I can just Google around to see if range-v3/yaml-cpp/Catch2 deps are available as Port or not.

It wouldn't make it any better for the port, as the Makefiles of the Port declare its dependencies. (See USES, LIB_DEPENDS and RUN_DEPENDS lines in the makefile).

For CMake based builds, it is best to pretty much only depend on the PkgConfig module and let it figure out compilation and linker flags etc. For aiding developers, having a `install-deps` script is perfect! that would even help on CI.

**Comment by christianparpart** [comment]
> > Would this script-based approach work for you (as in: FreeBSD)? I can just Google around to see if range-v3/yaml-cpp/Catch2 deps are available as Port or not.
> 
> It wouldn't make it any better for the port, as the Makefiles of the Port declare its dependencies. (See USES, LIB_DEPENDS and RUN_DEPENDS lines in the makefile).

I think it makes it a little better in a sense that we could actually:

1. get rid of CPM all together
2. decouple deps-fetching from CMake

You are not mandated to invoke scripts/install-deps.sh anyways. So in a Portfile (or rpm/deb) package creator does provide the dependencies just like you said, that's perfectly fine (and actually preferred by me anyways) :)

> For CMake based builds, it is best to pretty much only depend on the PkgConfig module and let it figure out compilation and linker flags etc. For aiding developers,

I will see what I can do to improve this dependency-situation. :)

> having a `install-deps` script is perfect! that would even help on CI.

I'm using it on CI for quickly installing the deps and on my machines whenever I need to quickly fetch'n'install them. So the existence of such a script is still valuable (for convenience reasons).

Now, let me decouple this ..... *zensored*.

**Comment by christianparpart** [comment]
For a more realtime off-github communication, you can join our Discord, if this is of any interest to you. :)

**Comment by herrhotzenplotz** [comment]
> For a more realtime off-github communication, you can join our Discord, if this is of any interest to you. :)

Sorry, I'm more of an IRC type person :neutral_face:

**Pull request merged**